<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RGenEDA Tutorial • RGenEDA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RGenEDA Tutorial">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RGenEDA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Snail1_Vignette.html">RGenEDA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RGenEDA Tutorial</h1>
            
      

      <div class="hidden name"><code>Snail1_Vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>RGenEDA</code> is designed to provide a streamlined, unified
and reproducible framework for exploratory data analysis across multiple
omics data types. This vignette introduces the key components of the
<code>RGenEDA</code> package using bulk RNA-seq data from the paper <a href="https://pubmed.ncbi.nlm.nih.gov/29729076/" class="external-link">Genomewide binding of
transcription factor Snail1 in triple-negative breast cancer cell</a>
(Maturi, et al. 2018). Raw counts and metadata were obtained from <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5244/Downloads" class="external-link">The
Gene Expression Atlas</a> under ENA:ERP019920, E-MTAB-5244.</p>
<p>The data has be pre-wrangled but the standard <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" class="external-link">deseq2
framework</a> will be applied here to demonstrate the functionality of
<code>RGenEDA</code>.</p>
<p>The paper explores epithelial cell line HS578T. The authors have
introduced a knock-out of the Snail1 transcription factor and compare
these with wild-type (WT) HS578T cells.</p>
<div class="section level3">
<h3 id="table-of-contents">Table of contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h3>
<ul>
<li><a href="#load-and-inspect-the-data">Load and inspect the
data</a></li>
<li><a href="#define-metadata">Define metadata</a></li>
<li><a href="#processing-and-normalization">Processing and
normalization</a></li>
<li><a href="#create-a-geneda-object">Create a GenEDA object</a></li>
<li><a href="#count-distributions-across-samples">Count distrubutions
across samples</a></li>
<li><a href="#sample-euclidean-distances-with-hierarchical-clustering">Sample
Eucliden distances with hierarchical clustering</a></li>
<li><a href="#identify-highly-variable-genes">Identify highly variable
genes</a></li>
<li><a href="#principal-component-analysis">Principal component
analysis</a></li>
<li><a href="#extract-and-visualize-pca-results">Extract and visualize
PCA results</a></li>
<li><a href="#explore-eigen-vectors-of-individual-pcs">Explore Eigen
vectors of individual PCs</a></li>
<li><a href="#correlate-pcs-with-metadata">Correlate PCs with
metadata</a></li>
<li><a href="#explore-degs">Explore DEGs</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="load-and-inspect-the-data">Load and inspect the data<a class="anchor" aria-label="anchor" href="#load-and-inspect-the-data"></a>
</h3>
<p>Data has been included in this package for your convenience. It can
be easily accessed using the following commands. Here, we have one key
variable of interest (Genotype) for simplicity sake (though your
experiment may have many. <code>RGenEDA</code> can handle any number of
covariates.)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Load the counts and metadata and extract</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Snail1KO"</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">Snail1KO</span><span class="op">[[</span><span class="st">"rawCounts"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">Snail1KO</span><span class="op">[[</span><span class="st">"metadata"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Sanity check</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="co">#&gt;          ERR1736465 ERR1736466 ERR1736467 ERR1736468 ERR1736469 ERR1736470</span></span>
<span><span class="co">#&gt; TSPAN6          434        829        334        779        578        602</span></span>
<span><span class="co">#&gt; TNMD              0          0          0          0          0          0</span></span>
<span><span class="co">#&gt; DPM1           2202       3461       1627       4241       3676       3946</span></span>
<span><span class="co">#&gt; SCYL3            45         72         31        119         69         92</span></span>
<span><span class="co">#&gt; C1orf112        320        558        235        516        468        417</span></span>
<span><span class="co">#&gt; FGR               0         10          0          0          0          0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="co">#&gt;                Sample          Disease Genotype        CellLine_CellType</span></span>
<span><span class="co">#&gt; ERR1736465 ERR1736465 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736466 ERR1736466 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736467 ERR1736467 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736468 ERR1736468 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736469 ERR1736469 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736470 ERR1736470 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span></code></pre></div>
<p>It is helpful to establish color palettes early on in an exploratory
analysis to keep figures consistent. <code>RGenEDA</code> uses the list
of named vectors convention for creating color vectors for plotting
functions.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Color palettes for plotting</span></span>
<span><span class="va">colorList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  Genotype <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span> <span class="op">=</span> <span class="st">"steelblue3"</span>, <span class="st">"Snai1_KO"</span> <span class="op">=</span> <span class="st">"firebrick3"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="processing-and-normalization">Processing and normalization<a class="anchor" aria-label="anchor" href="#processing-and-normalization"></a>
</h3>
<p>The standard DESeq2 workflow can now be applied. The end goal is to
obtain normalized counts and differential expression results. Lowly
expressed genes are filtered out before running <code>DESeq2.</code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Ensure tables are not scrambled</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span></span>
<span>  countData <span class="op">=</span> <span class="va">counts</span>,</span>
<span>  colData <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>  design <span class="op">=</span> <span class="op">~</span> <span class="va">Genotype</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span></span>
<span><span class="co">#&gt; design formula are characters, converting to factors</span></span>
<span></span>
<span><span class="co"># Set reference levels</span></span>
<span><span class="va">dds</span><span class="op">$</span><span class="va">Genotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">Genotype</span>, ref <span class="op">=</span> <span class="st">"WT"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pre-filter: keep genes with at least 10 counts in  at least 3 samples</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">3</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Run DESeq2</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rlog transform the data and extract normalized matrix</span></span>
<span><span class="va">rld</span> <span class="op">&lt;-</span> <span class="fu">rlog</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">rld</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-a-geneda-object">Create a GenEDA object<a class="anchor" aria-label="anchor" href="#create-a-geneda-object"></a>
</h3>
<p>With the normalized counts and metadata prepared, a
<code>GenEDA</code> object can be initialized. This object will store
all components of your analysis, starting with normalized data and
metadata (bare minimum requirements).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Factor metadata</span></span>
<span><span class="va">metadata</span><span class="op">$</span><span class="va">Genotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">Genotype</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">"Snai1_KO"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Initialize GenEDA object with normalized counts and metadata</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenEDA.html">GenEDA</a></span><span class="op">(</span></span>
<span>  normalized <span class="op">=</span> <span class="va">mat</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View object summary</span></span>
<span><span class="va">obj</span></span>
<span><span class="co">#&gt; geneda object</span></span>
<span><span class="co">#&gt;   features: 11247</span></span>
<span><span class="co">#&gt;   samples:  6</span></span>
<span><span class="co">#&gt;   HVGs: 0</span></span>
<span><span class="co">#&gt;   DimReduction: 0</span></span>
<span><span class="co">#&gt;   counts: NULL</span></span>
<span><span class="co">#&gt;   DEGs: NULL</span></span></code></pre></div>
<p>You will see that the object also contains slots for Highly Variable
Genes (HVGs), DimReduction (PCA, etc..) and Differentially Expressed
Genes (DEGs). These will be populated shortly with
<code>RGenEDA</code></p>
</div>
<div class="section level3">
<h3 id="count-distributions-across-samples">Count distributions across samples<a class="anchor" aria-label="anchor" href="#count-distributions-across-samples"></a>
</h3>
<p>To visualize normalized count distributions across samples, the
<code><a href="../reference/PlotCountDist.html">PlotCountDist()</a></code> function can be used. This is a quick and
helpful way to visualize effectiveness of normalization, as the overall
distributions should be similar across samples. Samples with very low or
very high overall counts compared to others might indicate problematic
samples, technical artifacts, or batch effect. This function returns a
<code>ggplot2</code> object to facilitate any additional
customization.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/PlotCountDist.html">PlotCountDist</a></span><span class="op">(</span><span class="va">obj</span>, split_by <span class="op">=</span> <span class="st">"Genotype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/count-dist-1.png" width="576" style="display: block; margin: auto;">
## Sample Eucliden distances with hierarchical clustering</p>
<p>To visualize replicate similarity, Euclidean distances between
samples can be calculated and plotted as a <code>pheatmap</code> heatmap
using the <code><a href="../reference/PlotDistances.html">PlotDistances()</a></code> function. Darker colors indicate
higher similarity, while lighter colors represent dissimilar samples.
This provides a quick assessment of replicate quality and metadata
features that drive clustering. To save <code>pheatmap</code> objects
within <code>RGenEDA</code>, we can use the <code><a href="../reference/GenSave.html">GenSave()</a></code>
function, which is similar in function to <code>ggsave()</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotDistances.html">PlotDistances</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  meta_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Genotype"</span><span class="op">)</span>,</span>
<span>  palettes <span class="op">=</span> <span class="va">colorList</span>,</span>
<span>  return <span class="op">=</span> <span class="st">"plot"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">hm</span><span class="op">$</span><span class="va">heatmap</span></span>
<span></span>
<span><span class="co"># GenSave(hm, "/path/to/EuclidenDistance_Heatmap.png", width = 6, height = 8)</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/clustering-1.png" width="60%" style="display: block; margin: auto;">
## Identify highly variable genes</p>
<p>Next, genes are ranked by decreasing variance to find highly variable
genes (HVGs) which most-likely drive biological differences. The full
variance profile curve for all genes profiled with
<code>plotHVGVariance()</code> as a means to pick a meaningful number of
HVGs to retain for calculating principal components in the next
section.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#----- Plot variance profile</span></span>
<span><span class="fu"><a href="../reference/PlotHVGVariance.html">PlotHVGVariance</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/hvgs-plot-1.png" width="60%" style="display: block; margin: auto;">
Based on the curve, 2,000 genes seems sufficient in capturing the
majority of variation. These genes can be extracted and retained in the
HVGs slot with <code><a href="../reference/FindVariableFeatures.html">FindVariableFeatures()</a></code></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#----- Add HVGs to object</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">obj</span>, <span class="fl">2000</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#== Access HVGs (2 methods)</span></span>
<span><span class="co"># 1: Accessor function</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/HVGs.html">HVGs</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "THY1"   "CCL2"   "SSX1"   "AKR1C2" "GPX1"   "MYL12B"</span></span>
<span></span>
<span><span class="co"># 2: Call the slot directly</span></span>
<span><span class="co">#head(obj@HVGs)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="principal-component-analysis">Principal component analysis<a class="anchor" aria-label="anchor" href="#principal-component-analysis"></a>
</h3>
<p>Using the identified HVGs, principal components can be calculated
using <code><a href="../reference/RunPCA.html">RunPCA()</a></code> This function stores PCA results in the
DimReduction slot, including:</p>
<p>• <code>$Loadings</code> (sample scores)</p>
<p>• <code>$Eigenvectors</code> (gene contributions)</p>
<p>• <code>$percent_var</code> (Percent variance explained per
component, up to PC5)</p>
<p>If <code><a href="../reference/FindVariableFeatures.html">FindVariableFeatures()</a></code> was not ran beforehand,
<code>RunPCA</code> will calculate HVGs by default with 2000 features.
This argument can be overriden directly using the <code>nfeatures</code>
argument.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; Calculating principal components from top 2000 HVGs</span></span>
<span><span class="co">#&gt; Percent variations:</span></span>
<span><span class="co">#&gt;       PC1       PC2       PC3       PC4       PC5 </span></span>
<span><span class="co">#&gt; "77.87 %" "11.93 %"  "4.52 %"  "3.36 %"  "2.32 %"</span></span>
<span></span>
<span><span class="co"># DimReductions can be accessed with accessor function</span></span>
<span><span class="co"># head(DimReduction(obj))</span></span>
<span></span>
<span><span class="co"># Inspect PCA outputs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">DimReduction</span><span class="op">$</span><span class="va">Loadings</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  PC1       PC2       PC3        PC4        PC5          PC6</span></span>
<span><span class="co">#&gt; ERR1736465 -5.845179  3.716576  1.620623 -1.0955893  0.1772979 3.594429e-14</span></span>
<span><span class="co">#&gt; ERR1736466 -6.433351 -1.186487  0.465169  2.4753119  0.2414526 3.577988e-14</span></span>
<span><span class="co">#&gt; ERR1736467 -6.592484 -2.136361 -2.110072 -1.3682232 -0.3897501 3.589042e-14</span></span>
<span><span class="co">#&gt; ERR1736468  6.063171 -2.271587  0.822507 -0.6827020  1.7746208 3.570818e-14</span></span>
<span><span class="co">#&gt; ERR1736469  6.007254 -1.272494  1.265257 -0.0950531 -1.9236799 3.560521e-14</span></span>
<span><span class="co">#&gt; ERR1736470  6.800590  3.150352 -2.063484  0.7662558  0.1200585 3.591129e-14</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">DimReduction</span><span class="op">$</span><span class="va">Eigenvectors</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  PC1         PC2           PC3          PC4          PC5</span></span>
<span><span class="co">#&gt; CENPI    0.003588589 -0.02172881 -0.0196200153 -0.015192540 -0.005752305</span></span>
<span><span class="co">#&gt; TULP3    0.021586498 -0.02096203  0.0134327078 -0.016091133  0.033236356</span></span>
<span><span class="co">#&gt; MAP3K14  0.021846399  0.04358180 -0.0447224066 -0.022827314 -0.064475180</span></span>
<span><span class="co">#&gt; AP3D1    0.026800954  0.02836574 -0.0380992712 -0.006296689  0.006278297</span></span>
<span><span class="co">#&gt; DIP2B   -0.002712758 -0.01559409 -0.0310157471 -0.028917713  0.008953505</span></span>
<span><span class="co">#&gt; TAZ      0.003357044  0.01302154 -0.0009023767 -0.003193418 -0.005208482</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">DimReduction</span><span class="op">$</span><span class="va">percent_var</span><span class="op">)</span></span>
<span><span class="co">#&gt;       PC1       PC2       PC3       PC4       PC5 </span></span>
<span><span class="co">#&gt; "77.87 %" "11.93 %"  "4.52 %"  "3.36 %"  "2.32 %"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-and-visualize-pca-results">Extract and visualize PCA results<a class="anchor" aria-label="anchor" href="#extract-and-visualize-pca-results"></a>
</h3>
<p>PCA results merged with metadata can easily be extracted using
<code><a href="../reference/ExtractPCA.html">ExtractPCA()</a></code> which enables flexible plotting. For
convenience, <code><a href="../reference/PlotPCA.html">PlotPCA()</a></code> allows a quick visualization which
can also be further customized with <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">pcaDF</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractPCA.html">ExtractPCA</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pcaDF</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  PC1       PC2       PC3        PC4        PC5          PC6</span></span>
<span><span class="co">#&gt; ERR1736465 -5.845179  3.716576  1.620623 -1.0955893  0.1772979 3.594429e-14</span></span>
<span><span class="co">#&gt; ERR1736466 -6.433351 -1.186487  0.465169  2.4753119  0.2414526 3.577988e-14</span></span>
<span><span class="co">#&gt; ERR1736467 -6.592484 -2.136361 -2.110072 -1.3682232 -0.3897501 3.589042e-14</span></span>
<span><span class="co">#&gt; ERR1736468  6.063171 -2.271587  0.822507 -0.6827020  1.7746208 3.570818e-14</span></span>
<span><span class="co">#&gt; ERR1736469  6.007254 -1.272494  1.265257 -0.0950531 -1.9236799 3.560521e-14</span></span>
<span><span class="co">#&gt; ERR1736470  6.800590  3.150352 -2.063484  0.7662558  0.1200585 3.591129e-14</span></span>
<span><span class="co">#&gt;                Sample          Disease Genotype        CellLine_CellType</span></span>
<span><span class="co">#&gt; ERR1736465 ERR1736465 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736466 ERR1736466 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736467 ERR1736467 breast carcinoma       WT HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736468 ERR1736468 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736469 ERR1736469 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span>
<span><span class="co">#&gt; ERR1736470 ERR1736470 breast carcinoma Snai1_KO HS578T - epithelial cell</span></span>
<span></span>
<span><span class="co"># Plot PCA</span></span>
<span><span class="fu"><a href="../reference/PlotPCA.html">PlotPCA</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>,</span>
<span>        x <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        y <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        color_by <span class="op">=</span> <span class="st">"Genotype"</span>,</span>
<span>        colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span> <span class="op">=</span> <span class="st">"steelblue3"</span>, <span class="st">"Snai1_KO"</span> <span class="op">=</span> <span class="st">"firebrick3"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">RGenEDA</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/extract-pca-1.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="explore-eigen-vectors-of-individual-pcs">Explore Eigen vectors of individual PCs<a class="anchor" aria-label="anchor" href="#explore-eigen-vectors-of-individual-pcs"></a>
</h3>
<p>The individual Eigenvectors (genes) that comprise a particular
component of interest can be extracted and their Z-scaled normalized
expression visualized as a heatmap annotated by the percent variation
explained using <code><a href="../reference/extractEigen.html">extractEigen()</a></code> and
<code><a href="../reference/PlotEigenHeatmap.html">PlotEigenHeatmap()</a></code>. This can again identify
sample-to-sample differences.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">pc1_eigen</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extractEigen.html">extractEigen</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>,</span>
<span>                          component <span class="op">=</span> <span class="st">"PC1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pc1_eigen</span><span class="op">)</span></span>
<span><span class="co">#&gt;      Gene  EigenVector       PctVar</span></span>
<span><span class="co">#&gt; 1   CENPI  0.003588589 0.0012877974</span></span>
<span><span class="co">#&gt; 2   TULP3  0.021586498 0.0465976905</span></span>
<span><span class="co">#&gt; 3 MAP3K14  0.021846399 0.0477265145</span></span>
<span><span class="co">#&gt; 4   AP3D1  0.026800954 0.0718291147</span></span>
<span><span class="co">#&gt; 5   DIP2B -0.002712758 0.0007359055</span></span>
<span><span class="co">#&gt; 6     TAZ  0.003357044 0.0011269741</span></span>
<span></span>
<span></span>
<span><span class="va">hm2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotEigenHeatmap.html">PlotEigenHeatmap</a></span><span class="op">(</span><span class="va">obj</span>,</span>
<span>                 pc <span class="op">=</span> <span class="st">"PC1"</span>,</span>
<span>                 n <span class="op">=</span> <span class="fl">25</span>,</span>
<span>                 annotate_by <span class="op">=</span> <span class="st">"Genotype"</span>,</span>
<span>                 annotate_colors <span class="op">=</span> <span class="va">colorList</span><span class="op">)</span></span>
<span><span class="va">hm2</span><span class="op">$</span><span class="va">heatmap</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/eigenvecs-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># GenSave(hm2, "/path/to/EuclidenDistance_Heatmap.png", width = 6, height = 8)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="correlate-pcs-with-metadata">Correlate PCs with metadata<a class="anchor" aria-label="anchor" href="#correlate-pcs-with-metadata"></a>
</h3>
<p>To interpret principal components, individual PCs can be correlated
with sample metadata using <code><a href="../reference/PlotEigenCorr.html">PlotEigenCorr()</a></code>. This function
computes Pearson correlations and displays them as a heatmap, helping to
reveal which metadata features are most associated with major axes of
variation. This function returns a list of 4 elements:</p>
<p>• <code>$cor_matrix</code> (Pearson correlation values)</p>
<p>• <code>$pval_matrix</code> (Associated correlation p-values)</p>
<p>• <code>$stars</code> (asterisk representations of p-values)</p>
<p>• <code>$plot</code> (Eigencorr plot, as a <code>ggplot2</code>
object, which can be saved with <code>ggsave</code>)</p>
<p><strong>Note:</strong> <code><a href="../reference/PlotOrdCorr.html">PlotOrdCorr()</a></code> can be used for
microbiome data as it correlates metadata features with NMDS beta values
rather than PCs.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">ec</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotEigenCorr.html">PlotEigenCorr</a></span><span class="op">(</span><span class="va">obj</span>, num_pcs <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="co">#&gt; Warning in cor(x, y): the standard deviation is zero</span></span>
<span><span class="va">ec</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/eigencorr-1.png" width="70%" style="display: block; margin: auto;">
## Explore DEGs</p>
<p>Differentially expressed genes can now be stored in the
<code>RGenEDA</code> object. <code>RGenEDA</code> was designed to
directly work with DEG tables derived from <code>DESeq2</code> and
contain the column names “baseMean”, “log2FoldChange” and “padj”. DEG
tables from other packages can be used with some minor dataframe
manipulation.</p>
<p>Note: multiple DEG assays can be appended to the DEGs slot by passing
an assay name in the <code><a href="../reference/SetDEGs.html">SetDEGs()</a></code> command (for example, raw
DESeq2 results and Shrunk DESeq2 results.)</p>
<p>DEG assays can be filterd and directly saved as a new assay in the
DEGs slot using <code><a href="../reference/FilterDEGs.html">FilterDEGs()</a></code></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set a new DEG assay</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDEGs.html">SetDEGs</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>, </span>
<span>               deg_table <span class="op">=</span> <span class="va">res</span>,</span>
<span>               assay <span class="op">=</span> <span class="st">"unfiltered"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter the DEG assay and save as a new one</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterDEGs.html">FilterDEGs</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>,</span>
<span>                  assay <span class="op">=</span> <span class="st">"unfiltered"</span>,</span>
<span>                  padj_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                  log2FC_thresh <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  saveAssay <span class="op">=</span> <span class="st">"padj05_lfc1"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Grab results with accessor function</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="../reference/DEGs.html">DEGs</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>, assay <span class="op">=</span> <span class="st">"unfiltered"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 11247</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="../reference/DEGs.html">DEGs</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">obj</span>, assay <span class="op">=</span> <span class="st">"padj05_lfc1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1198</span></span></code></pre></div>
<p>Basic DEG visualizations plots such as MA plots and volcano plots can
be plotted using <code><a href="../reference/PlotMA.html">PlotMA()</a></code> and <code><a href="../reference/PlotVolcano.html">PlotVolcano()</a></code>
“Num” and “Den” arguments refer to numerator level and denominator level
(reference level of experiment, in our case, “WT”)</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/PlotMA.html">PlotMA</a></span><span class="op">(</span><span class="va">obj</span>,</span>
<span>       assay <span class="op">=</span> <span class="st">"unfiltered"</span>,</span>
<span>       alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>       fc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">RGenEDA</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">size</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">size</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Warning: Removed 5 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 653 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 542 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/MA-1.png" width="70%" style="display: block; margin: auto;"></p>
<p>Or a volcano plot by specifying numerator and denominator
(denominator is your comparison reference level).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/PlotVolcano.html">PlotVolcano</a></span><span class="op">(</span><span class="va">obj</span>,</span>
<span>       assay <span class="op">=</span> <span class="st">"unfiltered"</span>, </span>
<span>       alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>       fc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>       den <span class="op">=</span> <span class="st">"WT"</span>,</span>
<span>       num <span class="op">=</span> <span class="st">"Snai1_KO"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Example Volcano"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 653 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 538 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="Snail1_Vignette_files/figure-html/Volcano-1.png" width="70%" style="display: block; margin: auto;">
To explore whether a differentially expressed gene (DEG) is also a
highly variable gene (HVG), the intersect of these two vectors can be
taken to obtain highly variable DEGs (hvDEGs), which can be useful for
applications such as gene-set variation analysis (GSVA). To obtain this,
the <code><a href="../reference/FindHVDEGs.html">FindHVDEGs()</a></code> function can be used. The direction of
fold change to intersect with HVGs can be either “positive”, or
“negative”, which returns a vector of genes, or “both” which contains a
list with <code>$positive</code>, <code>$negative</code>, and
<code>$both</code> slots.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/FindHVDEGs.html">FindHVDEGs</a></span><span class="op">(</span><span class="va">obj</span>, </span>
<span>           assay <span class="op">=</span> <span class="st">"padj05_lfc1"</span>,</span>
<span>           direction <span class="op">=</span> <span class="st">"negative"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 332 -log2FC hvDEGs found.</span></span>
<span><span class="co">#&gt;   [1] "CAMKK1"       "ACSM3"        "TMEM98"       "DLX6"         "MAP3K9"      </span></span>
<span><span class="co">#&gt;   [6] "PRSS21"       "FUZ"          "CEP68"        "ERCC1"        "BID"         </span></span>
<span><span class="co">#&gt;  [11] "ABCC2"        "DEF6"         "LRRC7"        "VCAN"         "SPATA7"      </span></span>
<span><span class="co">#&gt;  [16] "CAPG"         "MPC1"         "GUCY1B1"      "EYA2"         "ANKRD44"     </span></span>
<span><span class="co">#&gt;  [21] "PRKCQ"        "MTHFD2"       "TRO"          "NAV3"         "MAST4"       </span></span>
<span><span class="co">#&gt;  [26] "GAL"          "CLTCL1"       "DGCR2"        "MGAT4A"       "VASH1"       </span></span>
<span><span class="co">#&gt;  [31] "ST6GAL1"      "ZNF532"       "TUBE1"        "MLH1"         "ARHGEF1"     </span></span>
<span><span class="co">#&gt;  [36] "SLC1A3"       "OPHN1"        "CARMIL1"      "IGSF9B"       "CXCL2"       </span></span>
<span><span class="co">#&gt;  [41] "MMP2"         "TESC"         "DLL3"         "DSP"          "SYDE2"       </span></span>
<span><span class="co">#&gt;  [46] "PCSK5"        "TRMT2A"       "SMARCB1"      "PPM1F"        "SRRD"        </span></span>
<span><span class="co">#&gt;  [51] "SEPTIN3"      "SYNGR1"       "PCK2"         "BMP7"         "ISM1"        </span></span>
<span><span class="co">#&gt;  [56] "TRIB3"        "FERMT1"       "ZNF516"       "KLHL4"        "CPPED1"      </span></span>
<span><span class="co">#&gt;  [61] "FZD3"         "JPH1"         "GDAP1"        "TUBB4A"       "FCGRT"       </span></span>
<span><span class="co">#&gt;  [66] "NOVA2"        "OLFM2"        "TIMM50"       "DMAC2"        "CARD8"       </span></span>
<span><span class="co">#&gt;  [71] "TMEM59L"      "SCN1B"        "CADM4"        "PTN"          "GIMAP2"      </span></span>
<span><span class="co">#&gt;  [76] "TRIM14"       "KANK1"        "COL1A1"       "CLEC2B"       "PARP11"      </span></span>
<span><span class="co">#&gt;  [81] "VDR"          "ADGRD1"       "PHC1"         "SOBP"         "PHACTR2"     </span></span>
<span><span class="co">#&gt;  [86] "SEMA5A"       "GHR"          "PCDHB5"       "PCDHB6"       "CDH6"        </span></span>
<span><span class="co">#&gt;  [91] "SLC12A7"      "EHHADH"       "PODXL2"       "ARHGEF26"     "EFEMP1"      </span></span>
<span><span class="co">#&gt;  [96] "IL18R1"       "ID2"          "NOL10"        "SLC25A12"     "SDC1"        </span></span>
<span><span class="co">#&gt; [101] "ORC4"         "PAPPA2"       "RHOU"         "CTH"          "AGMAT"       </span></span>
<span><span class="co">#&gt; [106] "ADGRL2"       "IFT46"        "MREG"         "UBE3D"        "TGIF2"       </span></span>
<span><span class="co">#&gt; [111] "ABCG2"        "LGALSL"       "BCL11A"       "PCDHB12"      "RDH10"       </span></span>
<span><span class="co">#&gt; [116] "PDZRN3"       "KIAA1549"     "BAZ2B"        "PFKFB2"       "DAW1"        </span></span>
<span><span class="co">#&gt; [121] "PTGIS"        "SNAI1"        "ZNF576"       "BMP4"         "EML2"        </span></span>
<span><span class="co">#&gt; [126] "TMX4"         "ID1"          "SSX1"         "GNAZ"         "PODXL"       </span></span>
<span><span class="co">#&gt; [131] "CHAC1"        "KIF1A"        "AFDN"         "ZNF227"       "THOC6"       </span></span>
<span><span class="co">#&gt; [136] "PER2"         "TTC9"         "PSAT1"        "MSI1"         "ACVR1B"      </span></span>
<span><span class="co">#&gt; [141] "OS9"          "MAP7"         "NHSL1"        "NIBAN1"       "EDNRB"       </span></span>
<span><span class="co">#&gt; [146] "GATA4"        "CTSV"         "KIAA0319"     "FLOT1"        "SULF1"       </span></span>
<span><span class="co">#&gt; [151] "FXYD6"        "GPAT3"        "BMPR1B"       "LEF1"         "CLSTN3"      </span></span>
<span><span class="co">#&gt; [156] "INHBE"        "RAB15"        "STON2"        "NUDT7"        "CDH13"       </span></span>
<span><span class="co">#&gt; [161] "GAREM1"       "TTYH2"        "IGFBP4"       "GEMIN7"       "WTIP"        </span></span>
<span><span class="co">#&gt; [166] "CRABP2"       "ETNK2"        "ALDH1L1"      "OTULINL"      "TNFAIP8"     </span></span>
<span><span class="co">#&gt; [171] "FAXC"         "SDK1"         "ZNF214"       "NCAM1"        "FEZ1"        </span></span>
<span><span class="co">#&gt; [176] "MPP7"         "CDH8"         "IL18"         "AKR1C2"       "TDO2"        </span></span>
<span><span class="co">#&gt; [181] "SPOCK1"       "PRDM8"        "BMP6"         "PTPRD"        "KCTD15"      </span></span>
<span><span class="co">#&gt; [186] "ANKFN1"       "KIF5A"        "TSPAN7"       "CD109"        "MMP14"       </span></span>
<span><span class="co">#&gt; [191] "ARSL"         "TMSB15A"      "TSPAN33"      "ZNF208"       "FAM171A2"    </span></span>
<span><span class="co">#&gt; [196] "PDPN"         "CADM3"        "EIF4E3"       "RPL22L1"      "HHIP"        </span></span>
<span><span class="co">#&gt; [201] "TBX20"        "STEAP1"       "ELAPOR2"      "ZNF704"       "CHMP4C"      </span></span>
<span><span class="co">#&gt; [206] "ADCY1"        "SLC16A9"      "PKNOX2"       "JCAD"         "JAM3"        </span></span>
<span><span class="co">#&gt; [211] "STXBP4"       "PPFIBP2"      "CYB5R2"       "PRTG"         "RIMKLB"      </span></span>
<span><span class="co">#&gt; [216] "NNMT"         "GPRC5B"       "OR51B5"       "OR51I1"       "CYP2S1"      </span></span>
<span><span class="co">#&gt; [221] "NXN"          "DDIT4"        "COA6"         "ZNF30"        "STXBP6"      </span></span>
<span><span class="co">#&gt; [226] "CXCL8"        "ZFPM2"        "SMAD1"        "KRT8"         "HOXB9"       </span></span>
<span><span class="co">#&gt; [231] "CHD7"         "LURAP1"       "DLK2"         "LRRC34"       "CYP4F11"     </span></span>
<span><span class="co">#&gt; [236] "TLN2"         "ZNF556"       "CALB2"        "SLCO2A1"      "BRSK2"       </span></span>
<span><span class="co">#&gt; [241] "NUPR1"        "OR51B6"       "EID2"         "EID2B"        "SYNE3"       </span></span>
<span><span class="co">#&gt; [246] "NCKAP5"       "MAGEF1"       "SOX12"        "HNRNPA0"      "AP3S1"       </span></span>
<span><span class="co">#&gt; [251] "ZNF223"       "ARL14"        "NLRP11"       "F2R"          "ZNF322"      </span></span>
<span><span class="co">#&gt; [256] "OR52D1"       "FDX1P1"       "PLCB1"        "CADM1"        "OR51B4"      </span></span>
<span><span class="co">#&gt; [261] "TMEM121B"     "SCN5A"        "THAP7"        "OR51M1"       "SLC24A3"     </span></span>
<span><span class="co">#&gt; [266] "MRPL40"       "PRAME"        "KCNQ5"        "KRT10"        "ZNF397"      </span></span>
<span><span class="co">#&gt; [271] "AKR1C1"       "ZKSCAN4"      "ZNF70"        "PABIR1"       "OR51I2"      </span></span>
<span><span class="co">#&gt; [276] "FAM78B"       "ZFP92"        "AKR1C3"       "PPP1R26"      "POTEF"       </span></span>
<span><span class="co">#&gt; [281] "TCF4"         "ZNF512B"      "MVB12B"       "ZSCAN26"      "STMN3"       </span></span>
<span><span class="co">#&gt; [286] "COL13A1"      "TUBA3C"       "MDM4"         "FAM169A"      "ZNF830"      </span></span>
<span><span class="co">#&gt; [291] "SCAMP5"       "ZNF521"       "SFMBT2"       "DZIP3"        "OR2L2"       </span></span>
<span><span class="co">#&gt; [296] "GPANK1"       "MRPS18B"      "PRR3"         "TMEM200C"     "GPSM3"       </span></span>
<span><span class="co">#&gt; [301] "EMP2"         "RTL10"        "RABGAP1L-IT1" "RFPL4A"       "SNHG14"      </span></span>
<span><span class="co">#&gt; [306] "APCDD1L-DT"   "YWHAEP1"      "RNASEH1-AS1"  "ZSCAN31"      "FOXD2-AS1"   </span></span>
<span><span class="co">#&gt; [311] "POU5F1P4"     "PRKCQ-AS1"    "GUSBP2"       "PEG10"        "N4BP2L2"     </span></span>
<span><span class="co">#&gt; [316] "LINC00504"    "RBBP4P1"      "CHCHD10"      "PCDHB17P"     "PECAM1"      </span></span>
<span><span class="co">#&gt; [321] "GNAO1-AS1"    "SH3GL1P1"     "LIN37"        "NCBP2AS2"     "CAHM"        </span></span>
<span><span class="co">#&gt; [326] "ZNF595"       "FAM106A"      "H2BC3"        "H2AC16"       "H4C1"        </span></span>
<span><span class="co">#&gt; [331] "OR51B2"       "H3C3"</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Michael Martinez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
